### PR TITLE
fix(developer): bring kmc-package inline with file format spec 🗜

### LINKED ISSUE #9198: bug(developer): kmc-package needs to strip startMenuItem.location and icon if default 🗜
As seen in ahom_star:

```
JSON not equal startMenu>(length): 3 !== 2
{
  folder: 'ahom_star',
  addUninstallEntry: true,
  items: [
    {
      name: '(new)',
      filename: '(Product Configuration)',
      arguments: 'shom_star'
    },
    {
      name: '(new)',
      filename: 'Ahom_star.kvk',
      arguments: 'Ahom fonts'
    }
  ]
}
---
{
  folder: 'ahom_star',
  items: [
    {
      name: '(new)',
      fileName: '(Product Configuration)',
      arguments: 'shom_star',
      icon: '',
      location: 'psmelStartMenu'
    },
    {
      name: '(new)',
      fileName: 'Ahom_star.kvk',
      arguments: 'Ahom fonts',
      icon: '',
      location: 'psmelStartMenu'
    }
  ]
}
File kmp.json differs in output\17.0.107-alpha-local\ahom_star.kmp and output\kmc\ahom_star.kmp
```

### LINKED ISSUE #9203: bug(developer): kmc-package should not emit strings as they are not used in kmp.json 🗜
As seen in release/n/nailangs

### LINKED ISSUE #9204: bug(developer): kmc-package did not strip paths from `readmeFile` or `graphicFile` 🗜
As seen in release/gff/gff_amharic

### LINKED ISSUE #9205: feat(developer): kmc-package needs to support kmp.inf 🗜
To allow for downlevel versions of Keyman to load packages compiled by kmc-package, we need to fully support kmp.inf.

### LINKED ISSUE #9207: bug(developer): kmc-package needs to trim fields in kmp.json 🗜
As seen in release/rac/rac_marwari

### LINKED ISSUE #9208: chore(developer): read keyboard metadata when building package 🗜
Packages need to have keyboard metadata (e.g. version, rtl, etc) updated while they are being compiled. This applied to most packages in the repository.

### LINKED ISSUE #9209: bug(developer): kps-file had incorrect case for some fields 🗜

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9198.
Fixes #9203.
Fixes #9204.
Fixes #9205.
Fixes #9207.
Fixes #9208.
Fixes #9209.

Seven fixes. Four of these are in a single commit, as they were too tangled to easily do as separate commits. A thousand apologies.

* 306934eb62 addresses several issues with the package compiler:

  * refactors the reading of the metadata from keyboards to make use of it when refreshing the keyboard metadata in kmp.json (#9208). This is the bulk of the changes.
   * Fixes case on some fields in the kps file format (#9209)
   * Removes code which emitted strings table to kmp.json, as these were only ever used for package installer executables (#9203)
   * Cleans up code which emitted Start Menu items to make it match spec (#9198)

* 3773522e06 adds kmc-package support for writing kmp.inf, fixing #9205.

* 2085c26ab0 trims fields when emitting kmp.json, fixing #9207.

* 84c04d5cbd strips paths from `readmeFile` and `graphicFile`, fixing #9204.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-package/src/compiler/kmp-compiler.ts [hunk 1]
```diff
@@ -71,7 +71,11 @@ export class KmpCompiler {
     if(kps.options) {
       for (let [src,dst] of keys) {
         if (kps.options[src]) {
-          kmp.options[dst] = kps.options[src];
+          if(dst == 'graphicFile' || dst == 'readmeFile') {
+            kmp.options[dst] = /[/\\]?([^/\\]*)$/.exec(kps.options[src])[1];
```
Comment (jahorton) [inline developer/src/kmc-package/src/compiler/kmp-compiler.ts:81]: So... why are we stripping the path selectively here?  I assume there are other `dst` values that should remain unstripped?

File: common/web/types/src/package/kps-file.ts [hunk 2]
```diff
@@ -94,7 +94,7 @@ export interface KpsFileKeyboard {
   version: string;
   oSKFont?: string;
   displayFont?: string;
-  rtl?: boolean;
+  rTL?: string;
```
Comment (jahorton) [inline common/web/types/src/package/kps-file.ts:97]: What sort of convention is this?  I'd rather keep it purely lower-cased.  Also not a fan of `oSKFont`.  (Title-case for an abbrevation is... okay if necessary.  I'm generally not a fan of mixed-case within a coding abbreviation unless it causes issues parsing something in camelCase or similar.)

I checked the referenced issue, but there's no detail there for me to compare against.

Comment (jahorton) [reply common/web/types/src/package/kps-file.ts:97]: It sounds like there may be some pre-established spec that mandates this... but there was no link for me to follow and verify that.  I'll definitely acquiesce if this is the case.

File: developer/src/kmc-package/src/compiler/kmp-compiler.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/src/kmc-package/src/compiler/kmp-compiler.ts:81]: Those are the only two options that are paths

File: common/web/types/src/package/kps-file.ts (hunk 2 above)
Comment (mcdurdin) [reply common/web/types/src/package/kps-file.ts:97]: We've discussed this before. It's a "feature" of the xml2js implementation and options selected which I am still considering winding back, see #7353. I reckon we should call it dONKEYCase (or aSSCase but that's a little dangerous).

Comment (jahorton) [reply common/web/types/src/package/kps-file.ts:97]: Ah, so not exactly a "spec", but it's a pre-established thing we have to put up with for now b/c triage.  :+1:  Good to have the reference, at least.

**Review by jahorton — APPROVED** [review]

**Pull request merged**